File: FashionShop.Web/src/features/admin/orders/types/order.ts
export interface OrderSummary {
    orderId: string;
    orderDate: Date;
    fullName: string;
    phoneNumber: string;
    shippingAddress: string;
    shippingCommune: string;
    shippingDistrict: string;
    shippingCity: string;
    orderStatus: string;
    paymentMethod: string;
    paymentStatus: string;
    subTotal: number;
    totalItemCount: number;
    orderItems: OrderItemSummary[];
}

interface OrderItemSummary {
    imageUrl?: string;
}

export interface OrderDetail {
    orderId: string;
    orderDate: Date;
    fullName: string;
    phoneNumber: string;
    shippingAddress: string;
    shippingCommune: string;
    shippingDistrict: string;
    shippingCity: string;
    orderStatus: string;
    paymentMethod: string;
    paymentStatus: string;
    subTotal: number;
    shippingFee: number;
    discountAmount: number;
    totalAmount: number;
    note?: string;
    ShippingTrackingCode?: string;
    paymentDate?: Date;
    orderItems: OrderItemDetail[];
}

interface OrderItemDetail {
    orderItemId: number;
    productVariantId: string;
    productName: string;
    variantName: string;
    imageUrl?: string;
    unitPrice: number;
    quantity: number;
    totalLine: number;
    isReviewed: boolean;
}

File: FashionShop.Backend/FashionShop.Core/Entities/Order.cs
using FashionShop.Core.Common;
using FashionShop.Core.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FashionShop.Core.Entities
{
    [Table("Orders")]
    public class Order : BaseEntity
    {
        [Required]
        public Guid UserId { get; set; }

        public Guid? VoucherId { get; set; }

        [Required]
        [MaxLength(200)]
        public string ShippingAddress { get; set; }

        [Required]
        [MaxLength(200)]
        public string ShippingCommune { get; set; }

        [Required]
        [MaxLength(200)]
        public string ShippingDistrict { get; set; }

        [Required]
        [MaxLength(200)]
        public string ShippingCity { get; set; }

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal SubTotal { get; set; }

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal ShippingFee { get; set; } = 0;

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal DiscountAmount { get; set; } = 0;

        [Required]
        [Column(TypeName = "decimal(18,2)")]
        public decimal TotalAmount { get; set; }  // TotalAmount = SubTotal - DiscountAmount

        [Required]
        public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;

        [Required]
        public PaymentMethod PaymentMethod { get; set; } = PaymentMethod.COD;

        [Required]
        public PaymentStatus PaymentStatus { get; set; } = PaymentStatus.Unpaid;

        [MaxLength(500)]
        public string? Note { get; set; }

        [MaxLength(50)]
        [Column(TypeName = "varchar(50)")]
        public string? ShippingTrackingCode { get; set; }

        public DateTime OrderDate { get; set; } = DateTime.UtcNow;
        public DateTime? PaymentDate { get; set; }

        [ForeignKey("UserId")]
        public virtual User User { get; set; }

        [ForeignKey("VoucherId")]
        public virtual Voucher Voucher { get; set; }

        public virtual ICollection<OrderItem> OrderItems { get; set; } = new List<OrderItem>();

        public virtual ICollection<VoucherUsage> VoucherUsages { get; set; } = new List<VoucherUsage>();
    }
}
